Design/trivially_exportable_citations.md

```diff
@@ -0,0 +1,138 @@
+  std::string url;
+  std::string doi;
+  std::string bibtex;
+  std::string endnote;
+
+  void loadNexus(...);
+  void saveNexus(...);
+};
+```
+
+The fields here mirror those of [`NXcite`](http://download.nexusformat.org/doc/html/classes/base_classes/NXcite.html#nxcite).
+The following restrictions apply to ensure that in all cases somehting citable is allowed to exist:
+  - `description` is always optional (this isn't needed for citation, but gives insight as to why this citation is relevant)
+  - if `bibtex` is provided `endnote` must also be provided, and vice-versa (BibTex and Endnote contain essentially the same information, they can both be created if one can be. BibTex and Endnote do not imply a DOI is minted)
+  - if `doi` is provided, `url`, `bibtex` and `endnote` must all be provided (BibTex and Endnote can be generated from DOIs)
+  - if none of `doi`, `bibtex` or `endnote` are provided, `url` must be provided (there must be something there, even if this isn't citable a URL is better than nothing)
+
+Some helper "toString" methods may be beneficial.
+
+### Algorithm citations
+
+An additional `static` method will be added to `IAlgorithm` that returns a `std::vector<Citation const> const`.
+The developer has the option to overload this if the algorithm has relevant citations.
+By default the implementation in `IAlgorithm` will return an empty vector.
+
+### Citation recording
+
+- `WorkspaceHistory` and `AlgorithmHistory` shall be extended to include storage for framework and algorithm level citations respectively
+  - a `std::vector<Citation>` is likely sufficient
+- The requisite IO methods will be extended to handle loading and saving these citations to NeXus
+  - `WorkspaceHistory::loadNexus`
+  - `WorkspaceHistory::saveNexus`
+  - `AlgorithmHistory::saveNexus`
+
+Citations will be recorded in NeXus files using the [`NXcite`](http://download.nexusformat.org/doc/html/classes/base_classes/NXcite.html#nxcite) class.
+
+### Citation exporting
+
+There will be functionality to generate plain text lists of citations in all supported formats.
+This will be similar to the way we currently generate Python scripts form algorithm histories.
+
+A new algorithm `ExportCitations` shall be implemented.
+This algorithm takes a workspace as a property, gives the option of the desired export option and will output a file.
+
+The Algorithm History GUI may be extended to provide this functionality in a similar manner to Python scripts generated from the algorithm histories.
+
+### Documentation
+
+There are two main requirements for documentation integration:
+  - Listing the citations in the algorithm documentation page
+  - Allowing the documentation writer (developer) to cite the citations at specific points in the documentation
+
+The first point is trivial as we already do similar things for properties, related algorithms, etc.
+
+The second will involve more work.
+I don't have enough Sphinx know-how to suggest how this may be implemented.
```

Comment:
Do you want away to arbitrarily reference one of the citations or the whole lot?

Reply:
All of the citations for an algorithm should be listed in the corresponding algorithm documentation.

Ideally there should be a way to reference a citation for an algorithm within the RST documentation for that algorithm (that is the part I can't suggest how to implement here).

I'm not planning to make things citable from anywhere in the documentation, I'm not convinced there is a significant use case for that as you would likely want to cite the algorithm instead.
One exception may be framework specific things (such as MD event implementation).